eZ/Publish/Core/Repository/Repository.php

```diff
@@ -454,7 +459,40 @@ public function hasAccess($module, $function, APIUserReference $user = null)
             return $permissionSets;
         }
 
-        return false;// No policies matching $module and $function, or they contained limitations
+        // No policies matching $module and $function, or they contained limitations
+        return false;
+    }
+
+    /**
+     * Return true if at least one of the given policies overlaps $policy (has a wider scope).
+     * A policy can overlap other policy only if it has no limitations.
+     *
+     * @param \eZ\Publish\SPI\Persistence\User\Policy $policy
+     * @param \eZ\Publish\SPI\Persistence\User\Policy[] $rolePolicies
+     * @return bool
+     */
+    private function isOverlappedByWiderRolePolicy(Policy $policy, array $rolePolicies)
```

Comment:
Ok, got slightly better name now: _isOverlappedByPolicyOnSameRole_

But up to @pspania now